applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -1481,8 +1509,169 @@ void BaseSolidElement::CalculateConstitutiveVariables(
     // Setting the variables for the CL
     SetConstitutiveVariables(rThisKinematicVariables, rThisConstitutiveVariables, rValues, PointNumber, IntegrationPoints);
 
-    // Actually do the computations in the ConstitutiveLaw
+    // rotate to local axes strain/F
+    RotateToLocalAxes(rValues);
+
+    // Actually do the computations in the ConstitutiveLaw in local axes
     mConstitutiveLawVector[PointNumber]->CalculateMaterialResponse(rValues, ThisStressMeasure); //here the calculations are actually done
+
+    // We undo the rotation of strain/F, C, stress
+    RotateToGlobalAxes(rValues);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void InitialCheckLocalAxes(
+    const array_1d<double, 3>& rv1,
+    const array_1d<double, 3>& rv2,
+    const array_1d<double, 3>& rv3,
+    const double Tolerance = 1.0e4*std::numeric_limits<double>::epsilon()
+    )
+{
+    if (MathUtils<double>::Norm3(rv1) > 1.0 + Tolerance ||
+        MathUtils<double>::Norm3(rv2) > 1.0 + Tolerance ||
+        MathUtils<double>::Norm3(rv3) > 1.0 + Tolerance) {
+            KRATOS_ERROR << "The norm of one of the LOCAL_AXIS is greater than 1.0!" << std::endl;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void BuildRotationMatrix(
+    BoundedMatrix<double, 3, 3>& rRotationMatrix,
+    const array_1d<double, 3>& rv1,
+    const array_1d<double, 3>& rv2,
+    const array_1d<double, 3>& rv3
+    )
+{
+    rRotationMatrix(0, 0) = rv1[0]; rRotationMatrix(0, 1) = rv1[1]; rRotationMatrix(0, 2) = rv1[2];
+    rRotationMatrix(1, 0) = rv2[0]; rRotationMatrix(1, 1) = rv2[1]; rRotationMatrix(1, 2) = rv2[2];
+    rRotationMatrix(2, 0) = rv3[0]; rRotationMatrix(2, 1) = rv3[1]; rRotationMatrix(2, 2) = rv3[2];
+
+}
```

Comment:
why is this here?

I think it would make sense to move this to the StructuralMechanicsElementUtilities